rfcs/text/0030-linux-event-model.md

```diff
@@ -1736,6 +2063,29 @@ Stage 2: Identifies scope of impact of changes. Are breaking changes required? S
 The goal here is to research and understand the impact of these changes on users in the community and development teams across Elastic. 2-5 sentences each.
 -->
 
+### Ingestion mechanisms
+
+The Adaptive workload protections (AWP) sensor team, is working hard on integrating these new ECS fields into the endpoint agent codebase. The existing kprobe sensor tech will be upgraded to support much of the new widened process context defined in this RFC. At the same time eBPF sensor technology is being worked on and will eventually be the go to method for building these process events.
+
+### Usage mechanisms
+
+As outlined earlier in this document. The primary use for this data will be to drive the Session View (aka Terminal view, aka Sessionizer). Because of the hierarchical nature of this data, there are edge cases where some parts of the process tree are not captured, or have been filtered out (due to data volume concerns). This widenened process context (e.g entry_leader, session_leader, parent, group_leader) will allow the Session View to repair disjointed trees, and ensure it can represent the data as accurately as possible.
+
+Currently there are plans to integrate Session View into the kibana Timeline UI along side the Process Analyzer feature. There are also plans to add a "Sessions" tab to the Endpoint security hosts page.
+
+Another major win to this widenened process context is the ability to create more targeted rules. There are many situations where you may want to alert on a specific process executable, but only in certain cases. For example, perhaps it's ok to run the mysql cli client, but only if the process.entry_leader.user.name != 'root' and process.entry_leader.entry_meta.source.ip: <IP of a trusted bastion host>
+
+### ECS project
+
+As part of this RFC, there has been a few instances of a need to support arrays of fieldsets.
+
+The fields in question include:
+- process.file_descriptions (type: file_description)
+- process.previous (type: process)
+- group.supplemental (type: group)
+
+This will require involvement from the ECS team to add this support to the existing tooling.
```

Comment:
I don't know that we need to add it in here, but https://github.com/elastic/ecs/issues/1736 is the issue where I am working on this updated support.

Reply:
linked your issue here. ty